artifacts.go

```diff
@@ -0,0 +1,137 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package main
```

Comment:
Could we move this code under `util` so we can reuse it? Lets assume we have one day a tool `elastic-package` that allows us to validate a package and also build the `.tar.gz` file to upload manually to Kibana. Would be nice to share this code there. It will probably required to separate the handler code from the building the tar.gz code but I would assume this makes it easier to test if not already the case.

Reply:
I'm ok with moving it to a different place except utils, commons, shared, etc. ;)

Reply:
No strong opinion around the name. I assume this means we should also move the other `utils` code?

Reply:
The entire project can be adjusted to the standard layout: https://github.com/golang-standards/project-layout

Reply:
Moved to "archiver"

Reply:
SGTM to cleanup to follow the standard layout but lets do it in small steps.